ush/get_obs.py

```diff
@@ -0,0 +1,1004 @@
+    # the number of obs files needed, then we will need to retrieve files.
+    # In this case, set the sequence of hours corresponding to the archives
+    # from which files will be retrieved.
+    else:
+
+        arcv_hrs = [hr for hr in range(arcv_hr_start, arcv_hr_end+arcv_intvl_hrs, arcv_intvl_hrs)]
+        msg = dedent(f"""
+            At least some obs files needed needed for the current day (yyyymmdd_task)
+            do not exist on disk:
+                yyyymmdd_task = {yyyymmdd_task}
+            The number of obs files needed for the current day is:
+                num_files_needed = {num_files_needed}
+            The number of obs files that already exist on disk is:
+                num_existing_files = {num_existing_files}
+            Will retrieve remaining files by looping over archives corresponding to
+            the following hours (since hour 0 of the current day):
+                arcv_hrs = {arcv_hrs}
+            """)
+        logging.info(msg)
+    #
+    #-----------------------------------------------------------------------
+    #
+    # At this point, at least some obs files for the current day need to be
+    # retrieved.  Thus, loop over the relevant archives that contain obs for
+    # the day given by yyyymmdd_task and retrieve files as needed.
+    #
+    # Note that the NOHRSC data on HPSS are archived by day, with the archive
+    # for a given day containing 6-hour as well as 24-hour grib2 files.  As
+    # described above, the four 6-hour files are for accumulated snowfall at
+    # hour 0 of the current day (which represents accumulation over the last
+    # 6 hours of the previous day) as well as hours 6, 12, and 18, while the
+    # two 24-hour files are at hour 0 (which represents accumulation over all
+    # 24 hours of the previous day) and 12 (which represents accumulation over
+    # the last 12 hours of the previous day plus the first 12 hours of the
+    # current day).  Here, we will only obtain the 6-hour files.  In other
+    # workflow tasks, the values in these 6-hour files will be added as
+    # necessary to obtain accumulations over longer periods (e.g. 24 hours).
+    # Since the four 6-hour files are in one archive and are relatively small
+    # (on the order of kilobytes), we get them all with a single call to the
+    # retrieve_data.py script.
+    #
+    #-----------------------------------------------------------------------
+    #
+
+    # Whether to move the files or copy them from their raw to their processed
+    # locations.
+    mv_or_cp = 'cp'
+    # Whether to remove raw observations after processed directories have
+    # been created from them.
+    key = 'REMOVE_RAW_OBS_' + obtype
+    remove_raw_obs = vx_config[key]
+    # If the raw directories and files are to be removed at the end of this
+    # script, no need to copy the files since the raw directories are going
+    # to be removed anyway.
+    if remove_raw_obs:
+        mv_or_cp = 'mv'
```

Comment:
Mostly removed except for keeping the `remove_raw_obs` variable since it is used in a couple of places.  It makes it easier to change in case the variable name changes in the `config_defaults.yaml` file.